Sporadic failure of TestHealStartNStatusHandler on Windows CI

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
No sporadic failure in tests
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
Sporadic failure

```
[00:09:10] Time: 16:11:12 GMT 04/13/2018
[00:09:10] Error: Object not found: .minio.sys#buckets/mybucket/policy.json
[00:09:10]        1: cmd\xl-v1-object.go:832:cmd.xlObjects.DeleteObject()
[00:09:10]        2: cmd\xl-sets.go:594:cmd.(*xlSets).DeleteObject()
[00:09:10]        3: cmd\bucket-policy.go:156:cmd.removeBucketPolicy()
[00:09:10]        4: cmd\object-api-common.go:88:cmd.deleteBucketMetadata()
[00:09:10]        5: cmd\xl-sets.go:544:cmd.(*xlSets).DeleteBucket()
[00:09:10]        6: cmd\admin-handlers_test.go:255:cmd.(*adminXLTestBed).CleanupHealTestData()
[00:09:10]        7: cmd\admin-handlers_test.go:1208:cmd.TestHealStartNStatusHandler()
[00:09:10]        8: testing\testing.go:777:testing.tRunner()
[00:09:10] 
[00:09:10] API: SYSTEM()
[00:09:10] Time: 16:11:12 GMT 04/13/2018
[00:09:10] Error: Object not found: .minio.sys#buckets/mybucket/notification.xml
[00:09:10]        1: cmd\xl-v1-object.go:832:cmd.xlObjects.DeleteObject()
[00:09:10]        2: cmd\xl-sets.go:594:cmd.(*xlSets).DeleteObject()
[00:09:10]        3: cmd\object-api-common.go:155:cmd.removeNotificationConfig()
[00:09:10]        4: cmd\object-api-common.go:91:cmd.deleteBucketMetadata()
[00:09:10]        5: cmd\xl-sets.go:544:cmd.(*xlSets).DeleteBucket()
[00:09:10]        6: cmd\admin-handlers_test.go:255:cmd.(*adminXLTestBed).CleanupHealTestData()
[00:09:10]        7: cmd\admin-handlers_test.go:1208:cmd.TestHealStartNStatusHandler()
[00:09:10]        8: testing\testing.go:777:testing.tRunner()
[00:09:10] 
[00:09:10] API: SYSTEM()
[00:09:10] Time: 16:11:12 GMT 04/13/2018
[00:09:10] Error: Object not found: .minio.sys#buckets/mybucket/listener.json
[00:09:10]        1: cmd\xl-v1-object.go:832:cmd.xlObjects.DeleteObject()
[00:09:10]        2: cmd\xl-sets.go:594:cmd.(*xlSets).DeleteObject()
[00:09:10]        3: cmd\object-api-common.go:162:cmd.removeListenerConfig()
[00:09:10]        4: cmd\object-api-common.go:94:cmd.deleteBucketMetadata()
[00:09:10]        5: cmd\xl-sets.go:544:cmd.(*xlSets).DeleteBucket()
[00:09:10]        6: cmd\admin-handlers_test.go:255:cmd.(*adminXLTestBed).CleanupHealTestData()
[00:09:10]        7: cmd\admin-handlers_test.go:1208:cmd.TestHealStartNStatusHandler()
[00:09:10]        8: testing\testing.go:777:testing.tRunner()
[00:09:10] --- FAIL: TestHealStartNStatusHandler (0.57s)
[00:09:10]      admin-handlers_test.go:1194: did not find expected heal record in heal results
[00:09:10]      admin-handlers_test.go:1205: heal sequence stopped unexpectedly
```

<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Not sure 
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Unknown

--- a/cmd/posix.go
+++ b/cmd/posix.go
@@ -135,10 +135,10 @@ func getValidPath(path string) (string, error) {
 	if err != nil {
 		return path, err
 	}
+	defer os.Remove(pathJoin(path, ".writable-check.tmp"))
 	file.Close()
 
-	err = os.Remove(pathJoin(path, ".writable-check.tmp"))
-	return path, err
+	return path, nil
 }
 
 // isDirEmpty - returns whether given directory is empty or not.

